Sources/SafeDITool/SafeDITool.swift

```diff
@@ -234,3 +272,39 @@ extension String {
         #endif
     }
 }
+
+extension URL {
+    fileprivate init(relativePath: String) {
+        #if os(Linux)
+            self = URL(
+                fileURLWithPath: relativePath,
+                relativeTo: FileManager.default.currentDirectoryPath.asFileURL
+            )
+        #else
+            if #available(macOS 13.0, iOS 16.0, tvOS 16.0, watchOS 9.0, *) {
+                self = URL(
+                    filePath: relativePath,
+                    relativeTo: .currentDirectory()
+                )
+            } else {
+                self = URL(
+                    fileURLWithPath: relativePath,
+                    relativeTo: FileManager.default.currentDirectoryPath.asFileURL
+                )
+            }
+        #endif
+    }
+}
+
+protocol FileFinder {
+    func enumerator(
+        at url: URL,
+        includingPropertiesForKeys keys: [URLResourceKey]?,
+        options mask: FileManager.DirectoryEnumerationOptions,
+        errorHandler handler: ((URL, any Error) -> Bool)?
+    ) -> FileManager.DirectoryEnumerator?
+}
+
+extension FileManager: FileFinder {}
+
+var fileFinder: FileFinder = FileManager.default
```

Comment:
A little nuts to have a global like this, but since `SafeDITool` must be `Decodable` (which `FileFinder` most assuredly isn't) we can't add it as an ivar.

Open to better ideas here.

Reply:
Hmmm yeah this is a tricky one. I was thinking maybe you could wrap this in another type like `GlobalDependencies` or something but you still have the same setup. Not sure if there's anything that's obviously better.